tests/unit/modules/test_asset.py

```diff
@@ -13,25 +14,242 @@
 )
 
 
+class TestDoDiffer:
+    def test_equal(self):
+        assert asset.do_differ(
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                        "filters": ["a == a", "b == b"],
+                        "headers": {
+                            "foo": "bar"
+                        }
+                    },
+                    {
+                        "url": "http://def.com",
+                        "sha512": "def",
+                        "filters": ["d == d", "e == e"],
+                        "headers": {
+                            "bar": "foo"
+                        }
+                    },
+
+                ]
+            },
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://def.com",
+                        "sha512": "def",
+                        "filters": ["e == e", "d == d"],
+                        "headers": {
+                            "bar": "foo"
+                        }
+                    },
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                        "filters": ["b == b", "a == a"],
+                        "headers": {
+                            "foo": "bar"
+                        }
+                    },
+                ]
+            },
+        ) is False
+
+    def test_different_current_none(self):
+        assert asset.do_differ(
+            None,
+            {
+                "name": "a",
+                "builds": [
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                    }
+                ]
+            },
+        ) is True
+
+    def test_different(self):
+        assert asset.do_differ(
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://abc.com",
+                        "sha512": "abc",
+                    }
+                ]
+            },
+            {
+                "name": "asset",
+                "builds": [
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                    },
+                    {
+                        "url": "http://def.com",
+                        "sha512": "abc",
+                        "filters": ["abc == def"],
+                    }
+                ]
+            },
+        ) is True
+
+
+class TestDoBuildsDiffer:
```

Comment:
Would it be possible to merge tests in the `TestDoBuildsDiffer` class into the `TestDoDiffer` set of tests? There is a significant amount of overlap between those two sets of tests that we can safely remove. And then we can stop testing the private method ;)

See further comments for details.

Reply:
I don't feel very strongly about whether one should or shouldn't test private methods, so I'm good with either keeping `TestDoBuildsDiffer` or merging it with `TestDoDiffer` (while removing what's duplicated between them).  

I do love consistency though, and I thought I should take the same approach as we did with the [test suite for `role_utils`](https://github.com/sensu/sensu-go-ansible/blob/master/tests/unit/module_utils/test_role_utils.py#L12), where we test both private `_do_subjects_differ` method and `do_role_bindings_differ` as the only consumer of the same private method. But I do see now, that there's not as much overlap there :)

Reply:
As my original comment suggested, I am mostly concerned with the duplication since it makes any future changes even more painful.

As for the testing of private methods, this is just something that I try not to do since some linters (like pylint with default settings) yell at me ;) But in this case, pylint is tamed enough not to bark about this, so feel free to do whatever feels right to you.

Oh, and about that role utils tests: I must have missed that one ;) And now that I know about it, it does bother me a bit ;)